connect/transforms/src/main/java/org/apache/kafka/connect/transforms/Cast.java

```diff
@@ -80,7 +80,8 @@ public String toString() {
 
     private static final Set<Schema.Type> SUPPORTED_CAST_TYPES = EnumSet.of(
             Schema.Type.INT8, Schema.Type.INT16, Schema.Type.INT32, Schema.Type.INT64,
-                    Schema.Type.FLOAT32, Schema.Type.FLOAT64, Schema.Type.BOOLEAN, Schema.Type.STRING
+                    Schema.Type.FLOAT32, Schema.Type.FLOAT64, Schema.Type.BOOLEAN,
+                            Schema.Type.STRING, Schema.Type.BYTES
```

Comment:
Most of this looks fine, but I think just adding BYTES to this isn't quite right. The way this was originally written assumes that all types this transformation supports have a sensible translation to every other type (which is why it stuck to numbers, boolean, and strings). I think you're trying to add support for an additional set of casts that make sense (e.g. most things have a sensible `toString()`), but now the input and output are not symmetric -- some things make sense as an input, but not as an output (e.g. BYTES). So a single list might not be sufficient to do these checks correctly (e.g. as it is today, if you tried to do something like `bytes:string` conversion, instead of catching it as a config issue and giving an easy to understand error, it would be caught later and just say it encountered an unexpected type). If the symmetry is no longer going to be there, I think we need to separate out what we support as input and what we support as output (and I'm not even sure that will be sufficient, but it seems like the minimum we'd need).

Reply:
Makes sense. Let me check it out and push an update.